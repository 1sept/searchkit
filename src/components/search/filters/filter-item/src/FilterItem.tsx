import * as React from "react";

import {
	SearchkitComponent,
	FacetAccessor,
	FastClick,
	SearchkitComponentProps,
	ReactComponentType
} from "../../../../../core"

export interface FilterItemComponentProps {
	bemBlocks:{container:any, option:any},
	toggleFilter:Function,
	resetFilters:Function,
	translate:Function,
  selected:Boolean,
  label:string,
  count: number
}

export class FilterItemComponent extends React.Component<FilterItemComponentProps, any>{
	render(){
		const {bemBlocks, toggleFilter, translate, resetFilters, selected, label, count} = this.props
    const className = bemBlocks.option()
											.state({selected})
											.mix(bemBlocks.container("item"))

		return (
      <FastClick handler={toggleFilter} key={label}>
				<div className={className}>
					<div className={bemBlocks.option("text")}>{label}</div>
					<div className={bemBlocks.option("count")}>{count}</div>
				</div>
			</FastClick>
		)
	}
}
