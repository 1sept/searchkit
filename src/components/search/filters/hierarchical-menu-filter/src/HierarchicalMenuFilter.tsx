import * as React from "react";
import * as _ from "lodash";
import "../styles/index.scss";

import {
	SearchkitComponent,
  HierarchicalFacetAccessor,
	FastClick,
	SearchkitComponentProps
} from "../../../../../core"

export interface HierarchicalMenuFilterProps extends SearchkitComponentProps{
	id:string
	fields:Array<string>
	title:string
}

export class HierarchicalMenuFilter extends SearchkitComponent<HierarchicalMenuFilterProps, any> {
	public accessor:HierarchicalFacetAccessor

	static propTypes = _.defaults({
		id:React.PropTypes.string.isRequired,
		fields:React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
		title:React.PropTypes.string.isRequired
	}, SearchkitComponent.propTypes)

	constructor(props:HierarchicalMenuFilterProps) {
		super(props)
	}

	defineBEMBlocks() {
		var blockClass = this.props.mod || "hierarchical-menu";
		return {
			container:`${blockClass}-list`,
			option:`${blockClass}-option`
		};
	}

	defineAccessor() {
		return new HierarchicalFacetAccessor(
			this.props.id,
			{id:this.props.id, title:this.props.title, fields:this.props.fields, size:0}
		)
	}

	addFilter(option, level) {
		this.accessor.state = this.accessor.state.toggleLevel(
			level,option.key)

		this.searchkit.performSearch()
	}

	renderOption(level, option) {

		var block = this.bemBlocks.option

		var className = block().state({
			selected:this.accessor.state.contains(level, option.key)
		})

		return (
			<div key={option.key}>
				<FastClick handler={this.addFilter.bind(this, option,level)}>
					<div className={className}>
						<div className={block("text")}>{this.translate(option.key)}</div>
						<div className={block("count")}>{option.doc_count}</div>
					</div>
				</FastClick>
					{(() => {
						if(this.accessor.resultsState.contains(level,option.key)) {
							return this.renderOptions(level+1);
						}
					})()}
			</div>
		)
	}

	renderOptions(level) {
		let block = this.bemBlocks.container;
		return (
			<div className={block("hierarchical-options")}>
			{_.map(this.accessor.getBuckets(level), this.renderOption.bind(this,level))}
			</div>
		)
	}

  render(){
		let block = this.bemBlocks.container;
    return (
			<div className={block().mix(`filter--${this.props.id}`)}>
				<div className={block("header")}>{this.props.title}</div>
				<div className={block("root")}>
					{this.renderOptions(0)}
				</div>
			</div>
		)
	}

}
