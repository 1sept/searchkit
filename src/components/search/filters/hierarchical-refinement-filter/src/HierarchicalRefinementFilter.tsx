import * as React from "react";
import * as _ from "lodash";
import "../styles/index.scss";

import {
	SearchkitComponent,
  Accessor,
  LevelState,
	FastClick,
	AggsList,
	Terms,
	BoolMust,
	Term,
	NestedFilter
} from "../../../../../core"

export interface PathFacetAccessorOptions {
	field:string
	id:string
	title:string
}

export class PathFacetAccessor extends Accessor<LevelState> {
	state = new LevelState()
	options:any

	constructor(key, options:PathFacetAccessorOptions){
		super(key, options.id)
		this.options = options
	}

	getBuckets(level) {
		const results = this.getResults()
    const rpath = ['aggregations',this.key, "lvl"+level,"parents",'buckets']
    return _.get(results, rpath, [])
	}

	buildSharedQuery(query) {

		let levelFilters = this.state.getValue()
		let lastIndex = levelFilters.length - 1
		var filterTerms = _.map(levelFilters, (filter,i) => {

			let value = filter[0]
			let isLeaf = i === lastIndex
			let subField = isLeaf ? ".value" : ".ancestors"
			return Term(this.options.field + subField, value, {
				$name:this.options.title || this.translate(this.key),
				$value:this.translate(value),
				$id:this.options.id,
				$disabled: !isLeaf,
				$remove:()=> {
					this.state = this.state.clear(i)
		    }
			})
		})

		if(filterTerms.length > 0){
      query = query.addFilter(this.options.field,
				NestedFilter(this.options.field, BoolMust(filterTerms)))
    }

    return query
  }

  buildOwnQuery(query){

		let aggs = {
			"lvl0":{
				filter: {
					bool:{
						must:[
							Term("taxonomy.level", 1)
						]
					}
				},
				"aggs":{
					"parents":{
						 "terms":{
									"field":"taxonomy.value",
									"size":0
							}
					}
				}
			}
		}

		let levels = this.state.getValue()
		_.each(levels, (level,i) => {
			let ancestors = _.map(_.take(levels, i+1), (level)=>{
				return Term("taxonomy.ancestors", level[0])
			})
			aggs["lvl"+(i+1)] = {
				filter: {
					bool:{
						must:[
							Term("taxonomy.level", i+2),
							...ancestors
						]
					}
				},
				"aggs":{
					"parents":{
						 "terms":{
									"field":"taxonomy.value",
									"size":0
							}
					}
				}
			}
		})

		query = query.setAggs({
			taxonomy: {
				nested: {
					path: "taxonomy"
				},
				aggs: aggs
			}
		})

    return query
  }

}

export interface IHierarchicalRefinementFilter {
	id:string
	field:string
	title:string
	mod?:string
}

export class HierarchicalRefinementFilter extends SearchkitComponent<IHierarchicalRefinementFilter, any> {
	public accessor:PathFacetAccessor

	constructor(props:IHierarchicalRefinementFilter) {
		super(props)
	}

	defineBEMBlocks() {
		var blockClass = this.props.mod || "hierarchical-refinement";
		return {
			container:`${blockClass}-list`,
			option:`${blockClass}-option`
		};
	}

	shouldCreateNewSearcher() {
		return true;
	}

	defineAccessor() {
		return new PathFacetAccessor(
			this.props.id,
			{id:this.props.id, title:this.props.title, field:this.props.field}
		)
	}

	addFilter(level, option) {
		this.accessor.state = this.accessor.state.clear(level);
		this.accessor.state = this.accessor.state.add(level, option.key)
		this.searchkit.performSearch()
	}

	renderOption(level, option) {

		var block = this.bemBlocks.option
		var isSelected = this.accessor.resultsState.contains(level, option.key)

		var className = block().state({
			selected:isSelected
		})

		return (
			<div key={option.key}>
				<FastClick handler={this.addFilter.bind(this, level, option)}>
					<div className={className}>
						<div className={block("text")}>{this.translate(option.key)}</div>
						<div className={block("count")}>{option.doc_count}</div>
					</div>
				</FastClick>
					{(() => {
						if(isSelected) {
							return this.renderOptions(level+1);
						}
					})()}
			</div>
		)
	}

	renderOptions(level) {
		let block = this.bemBlocks.container;
		return (
			<div className={block("hierarchical-options")}>
			{
        _.map(this.accessor.getBuckets(level), this.renderOption.bind(this, level))
      }
			</div>
		)
	}

  render(){
		let block = this.bemBlocks.container;
    return (
			<div className={block().mix(`filter--${this.props.id}`)}>
				<div className={block("header")}>{this.props.title}</div>
				<div className={block("root")}>
					{this.renderOptions(0)}
				</div>
			</div>
		)
	}



}
