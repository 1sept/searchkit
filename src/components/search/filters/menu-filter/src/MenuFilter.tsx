import * as React from "react";
import * as _ from "lodash";
import * as classNames from 'classnames';
import {
	Searcher,
	SearchkitManager,
	SearchkitComponent,
	FacetAccessor
} from "../../../../../core"

require("./../styles/index.scss");

export interface IMenuFilter {
	field:string
	title:string
	id:string
}

export class MenuFilter extends SearchkitComponent<IMenuFilter, any> {
	accessor:FacetAccessor

	shouldCreateNewSearcher() {
		return true;
	}

	defineAccessor() {
		return new FacetAccessor(
			this.props.field,
			{id:this.props.id, operator:"OR", title:this.props.title}
		)
	}

	addFilter(option) {
		this.accessor.state = this.accessor.state.clear();
		if (option != "all") {
			this.accessor.state = this.accessor.state.add(option.key);
		}
		this.searchkit.performSearch()
	}

	renderOption(option) {

		let optionClassName = classNames({
			"menu-list-options__item":true,
			"menu-list-option":true,
			"menu-list-option--checked":this.accessor.state.contains(option.key)
		})

		return (
			<div className={optionClassName} key={option.key} onClick={this.addFilter.bind(this, option)}>
				<div className="menu-list-option__text">{option.key}</div>
			</div>
		)
	}

	renderAllOption() {
		let optionClassName = classNames({
			"menu-list-options__item":true,
			"menu-list-option":true,
			"menu-list-option--checked":!this.accessor.resultsState.getValue()
		})

		return (
			<div className={optionClassName} key="all" onClick={this.addFilter.bind(this, "all")}>
				<div className="menu-list-option__text">All</div>
			</div>
		)
	}

	render() {
		var className = classNames({
			"menu-list":true,
			[`filter--${this.props.id}`]:true
		})
		return (
			<div className={className}>
				<div className="menu-list-options">
				{this.renderAllOption()}
				{_.map(this.accessor.getBuckets(), this.renderOption.bind(this))}
				</div>
			</div>
		);
	}
}
