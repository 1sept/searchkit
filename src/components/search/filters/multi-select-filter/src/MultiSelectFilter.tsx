import * as React from "react";
import ESClient from "../../../../../domain/ESClient.ts";
import ElasticAccessors from "../../../../../domain/accessors/ElasticAccessors.ts"
import * as _ from "lodash";
import * as classNames from 'classnames';
import {StateAccessorRef} from "../../../../../domain/StateAccessors.ts"

require("./../styles/index.scss");

interface IMultiSelectFilter {
	searcher:ESClient;
	field:string
	operator?:string
}

export default class MultiSelectFilter extends React.Component<IMultiSelectFilter, any> {
	accessor:StateAccessorRef

	constructor(props:IMultiSelectFilter) {
		super(props)
		this.setAggs();
		this.accessor = this.props.searcher.accessors.registerAccessor(
			this.props.field,
			ElasticAccessors.facetFilter
		)
	}

	setAggs() {
		// this.props.searcher.setAggs(this.props.field, {
    //   "filter":{
    //     "match_all": {}
    //   },
		// 	"aggs": {
    //     [this.props.field]:{
    //       "terms":{"field":this.props.field}
    //     }
    //   }
		// })
	}

	addFilter(option) {
		this.accessor.toggle(option.key)
	}

	renderOption(option) {
		let className = classNames({
			"option__checkbox":true,
			"option__checkbox--checked":this.accessor.contains(option.key)
		})

		return (
			<div className="option" key={option.key} ref={option.key} onClick={this.addFilter.bind(this, option)}>
				<div className={className}></div>
				<div className="option__text">{option.key}</div>
				<div className="option__count">{option.doc_count}</div>
			</div>
		)
	}

	render() {
		return (
			<div className="refinement-list-filter">
				<div className="refinement-list-filter__header">{this.props.field}</div>
				<div className="refinement-list-filter__options">
			{(() => {
				if (_.has(this.props.searcher.results, `aggregations.${this.props.field}.buckets`)) {
	        return _.map(this.props.searcher.results.aggregations[this.props.field].buckets, this.renderOption.bind(this))
				}
			})()}
				</div>
      </div>
		);
	}
}
