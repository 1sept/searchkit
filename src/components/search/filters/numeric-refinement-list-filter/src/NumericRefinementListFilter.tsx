import * as React from "react";
import * as _ from "lodash";
import "../styles/index.scss";

import {
	SearchkitManager,
	SearchkitComponent,
	NumericOptionsAccessor,
	NumericOptions,
	FastClick
} from "../../../../../core"


export class NumericRefinementListFilter extends SearchkitComponent<NumericOptions, any> {
	accessor:NumericOptionsAccessor

	shouldCreateNewSearcher() {
		return true;
	}

	defineAccessor() {
		return new NumericOptionsAccessor(
			this.props.id,
			{id:this.props.id, field:this.props.field, options:this.props.options, title:this.props.title}
		)
	}

	defineBEMBlocks() {
		var blockName = this.props.mod || "numeric-refinement-list"
		return {
			container: blockName,
			option: `${blockName}-option`
		}
	}

	addFilter(option) {
		this.accessor.state = this.accessor.state.toggle(option.key)
		this.searchkit.performSearch()
	}

	isSelected(option) {
		return this.accessor.state.getValue() == option.key;
	}

	renderOption(option) {

		let block = this.bemBlocks.option
		let className = block()
			.mix(this.bemBlocks.container("item"))
			.state({
				selected:this.isSelected(option)
			})

		return (
			<FastClick handler={this.addFilter.bind(this, option)} key={option.key}>
				<div className={className}>
					<div className={block("text")}>{this.translate(option.key)}</div>
					<div className={block("count")}>{option.doc_count}</div>
				</div>
			</FastClick>
		)
	}

	render() {
		var block = this.bemBlocks.container
		var className = block().mix(`filter--${this.props.id}`)

		return (
			<div className={className}>
				<div className={block("header")}>{this.props.title}</div>
				<div className={block("options")}>
				{_.map(this.accessor.getBuckets(), this.renderOption.bind(this))}
				</div>
			</div>
		);
	}
}
