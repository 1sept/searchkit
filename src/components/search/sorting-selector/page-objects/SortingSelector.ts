import {Component, Input, Button, field, defaults} from "xenon";

export class SortingSelector extends Component {

  
}
