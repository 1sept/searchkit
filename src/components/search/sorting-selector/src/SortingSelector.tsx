import * as React from "react";
import * as _ from "lodash";
import "../styles/index.scss";

import {
	SearchkitComponent,
	SortingAccessor,
	FastClick,
	SortingOptions,
	SearchkitComponentProps,
	SortingOption
} from "../../../../core"

export interface SortingProps extends SearchkitComponentProps {
	options:Array<SortingOption>
}


export class SortingSelector extends SearchkitComponent<SortingProps, any> {
	accessor:SortingAccessor
	
	static propTypes = _.defaults({
		options:React.PropTypes.arrayOf(
			React.PropTypes.shape({
				label:React.PropTypes.string.isRequired,
				field:React.PropTypes.string.isRequired,
				order:React.PropTypes.string.isRequired
			})
		)
	}, SearchkitComponent.propTypes)


	defineAccessor() {
    return new SortingAccessor("sort", {options:this.props.options})
	}

	defineBEMBlocks() {
		return {
			container: (this.props.mod || "sorting-selector")
		}
	}

	renderOption(option) {
		return (
			<option key={option.label} value={option.label}>{option.label}</option>
		)
	}

	updateSorting(e) {
		let val:string = e.target.value;
		this.accessor.state = this.accessor.state.setValue(val);
		this.searchkit.performSearch();
	}

	getSelectedValue():string {
		return `${this.accessor.state.getValue()}`
	}

  render() {
    if(this.hasHits()){
			return (
				<div className={this.bemBlocks.container()}>
	      	<select onChange={this.updateSorting.bind(this)} value={this.getSelectedValue()}>
						{_.map(this.props.options, this.renderOption.bind(this))}
					</select>
	      </div>
			)
		}
		return null
  }
}
