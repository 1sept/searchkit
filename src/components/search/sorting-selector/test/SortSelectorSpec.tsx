import * as React from "react";
import {mount} from "enzyme";
import {SortingSelector} from "../src/SortingSelector";
import {SearchkitManager } from "../../../../core";
import {Toggle} from "../../../ui";
import * as ReactTestRenderer from 'react-test-renderer'
const bem = require("bem-cn");
import * as _ from "lodash"
import {
  fastClick, hasClass, jsxToHTML, printPrettyHtml
} from "../../../__test__/TestHelpers"

describe("SortingSelector tests", () => {

  beforeEach(()=> {
    this.searchkit = SearchkitManager.mock()
    spyOn(this.searchkit, "performSearch")
    this.setWrapper = (isEnzyme) => {
      const c = (
        <SortingSelector searchkit={this.searchkit} options={[
          {label:"Relevance"},
          {label:"Latest Releases", field:"released", order:"desc"},
          {label:"Earliest Releases", field:"released", order:"asc", key:"earliest"}
        ]} translations={{"Relevance":"Relevance translated"}}/>
      )
      this.wrapper = isEnzyme ? mount(c) : ReactTestRenderer.create(c)
      this.accessor = this.searchkit.accessors.accessors[0]
    }
    this.setResults = ()=> {
      this.searchkit.setResults({
        hits:{
          hits:[1,2],
          total:2
        }
      })
    }
  })

  test("is disabled when no results", ()=> {
    this.setWrapper()
    expect(this.wrapper).toMatchSnapshot()
  })

  test("renders with results", ()=> {
    this.setWrapper()
    this.setResults()
    expect(this.wrapper).toMatchSnapshot()
  })

  test("renders with selected value", ()=> {
    this.setWrapper()
    this.accessor.state = this.accessor.state.setValue("released_desc")
    this.setResults()
    this.setWrapper()
    expect(this.wrapper).toMatchSnapshot()
  })

  test("renders with defaultOption", ()=> {
    this.setWrapper()
    this.accessor.options.options[2].defaultOption = true
    this.setResults()
    this.setWrapper()
    expect(this.wrapper).toMatchSnapshot()
  })

  test("select new sort option", ()=> {
    this.setWrapper(true)
    this.accessor.state = this.accessor.state.setValue("released_desc")
    this.setResults()
    let earlyOption = this.wrapper.find("select").children().at(2)
    earlyOption.simulate("change")
    expect(this.accessor.state.getValue()).toBe("earliest")
    expect(this.searchkit.performSearch).toHaveBeenCalled()
  })

  test("handle prop reload without breaking computed keys", ()=> {
    this.setWrapper(true)
    this.wrapper.setProps({options:[
      {label:"Relevance"},
      {label:"Latest Releases", field:"released", order:"desc"},
      {label:"Earliest Releases", field:"released", order:"asc", key:"earliest"}
    ]})
    this.setResults()
    expect(this.wrapper.find('.sk-select')).toMatchSnapshot()
  })

  test("custom mod, className, listComponent", ()=> {
    this.wrapper = ReactTestRenderer.create(
      <SortingSelector searchkit={this.searchkit}
        mod="my-select" className="custom-class" listComponent={Toggle} options={[
        {label:"Relevance"},
        {label:"Latest Releases", field:"released", order:"desc"},
        {label:"Earliest Releases", field:"released", order:"asc", key:"earliest"}
      ]}/>
    )
    expect(this.wrapper).toMatchSnapshot()

  })

})
