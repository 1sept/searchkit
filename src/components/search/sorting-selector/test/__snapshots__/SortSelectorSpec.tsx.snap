// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingSelector tests custom mod, className, listComponent 1`] = `
<div
  className="my-select custom-class is-disabled"
  data-qa="options"
>
  <div
    className="my-select-option my-select__item is-active"
    data-key="none"
    data-qa="option"
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <div
      className="my-select-option__text"
      data-qa="label"
    >
      Relevance
    </div>
  </div>
  <div
    className="my-select-option my-select__item"
    data-key="released_desc"
    data-qa="option"
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <div
      className="my-select-option__text"
      data-qa="label"
    >
      Latest Releases
    </div>
  </div>
  <div
    className="my-select-option my-select__item"
    data-key="earliest"
    data-qa="option"
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <div
      className="my-select-option__text"
      data-qa="label"
    >
      Earliest Releases
    </div>
  </div>
</div>
`;

exports[`SortingSelector tests handle prop reload without breaking computed keys 1`] = `
<div
  className="sk-select"
>
  <select
    onChange={[Function]}
    value="none"
  >
    <option
      value="none"
    >
      Relevance translated
    </option>
    <option
      value="released_desc"
    >
      Latest Releases
    </option>
    <option
      value="earliest"
    >
      Earliest Releases
    </option>
  </select>
</div>
`;

exports[`SortingSelector tests is disabled when no results 1`] = `
<div
  className="sk-select is-disabled"
>
  <select
    onChange={[Function]}
    value="none"
  >
    <option
      disabled={undefined}
      value="none"
    >
      Relevance translated
    </option>
    <option
      disabled={undefined}
      value="released_desc"
    >
      Latest Releases
    </option>
    <option
      disabled={undefined}
      value="earliest"
    >
      Earliest Releases
    </option>
  </select>
</div>
`;

exports[`SortingSelector tests renders with defaultOption 1`] = `
<div
  className="sk-select"
>
  <select
    onChange={[Function]}
    value="earliest"
  >
    <option
      disabled={undefined}
      value="none"
    >
      Relevance translated
    </option>
    <option
      disabled={undefined}
      value="released_desc"
    >
      Latest Releases
    </option>
    <option
      disabled={undefined}
      value="earliest"
    >
      Earliest Releases
    </option>
  </select>
</div>
`;

exports[`SortingSelector tests renders with results 1`] = `
<div
  className="sk-select"
>
  <select
    onChange={[Function]}
    value="none"
  >
    <option
      disabled={undefined}
      value="none"
    >
      Relevance translated
    </option>
    <option
      disabled={undefined}
      value="released_desc"
    >
      Latest Releases
    </option>
    <option
      disabled={undefined}
      value="earliest"
    >
      Earliest Releases
    </option>
  </select>
</div>
`;

exports[`SortingSelector tests renders with selected value 1`] = `
<div
  className="sk-select"
>
  <select
    onChange={[Function]}
    value="released_desc"
  >
    <option
      disabled={undefined}
      value="none"
    >
      Relevance translated
    </option>
    <option
      disabled={undefined}
      value="released_desc"
    >
      Latest Releases
    </option>
    <option
      disabled={undefined}
      value="earliest"
    >
      Earliest Releases
    </option>
  </select>
</div>
`;
